### ISSUE TITLE
Netdata looping over accept

### ISSUE THREAD

**Issue by falbanese** [firehol/netdata#1503]
My platform is the following:
    admin@RT-N66U:/usr/lib# uname -a
    Linux RT-N66U 2.6.22.19 #1 Fri Nov 11 16:42:24 EST 2016 mips GNU/Linux
Netdata is installed as [entware-ng package](https://github.com/openwrt/packages/tree/ba20f0e4f7dfca8743c96bd0f4dc6c791dd8c4fe/admin/netdata).
When netdata starts, CPU utilization goes up to 100% and it is not possible to establish an HTTP connection. Digging into syslog, I find plenty of errors like the following:

    13824014: Cannot accept new incoming connection.

So, I suppose [this](https://github.com/firehol/netdata/blob/e19a4bde0a748b18339483df2a03a45430dbec4c/src/web_client.c#L62) call keeps on failing, but without a dump of ERRNO I cannot understand why.

Could you give me further assistance to debug this issue ?

**Comment by ktsaou** [comment]
The `error()` call in netdata prints out `errno`: https://github.com/firehol/netdata/blob/e19a4bde0a748b18339483df2a03a45430dbec4c/src/log.c#L314-L317

As you can see in the code, if it is not printed, then `errno == 0`.

**Comment by ktsaou** [comment]
Check also this: `accept4` is implemented by me if it is not available in your system: https://github.com/firehol/netdata/blob/2c9e84f0f17ae77d8a4939b3bcb9c3be63604da4/src/web_server.c#L50-L83

Can you check the `./configure` output if you have `socket4`?

**Comment by falbanese** [comment]
At the moment the toolchain is broken, I'll get back ASAP.

**Comment by falbanese** [comment]
I got to fix the cross-compilation toolchain.
Using the standard mipsel configuration, `configure` reports `accept4` as present.
I include the compilation [log.txt](https://github.com/firehol/netdata/files/687707/log.txt) for sake of completeness.

**Comment by ktsaou** [comment]
ok, so you tried to run and still have the same issue?

If `accept4()` is failing without `errno` I think autoconf is wrong. Your system does not have a working `accept4()`.

Could you please remove this line: https://github.com/firehol/netdata/blob/b1c742125df4e8af9969d1b0e9b0737e722c4b6e/configure.ac#L37

and retry? (you will need to run `./configure` again for that to make a change).

**Comment by ktsaou** [comment]
hm... I am now sure this is the case. `man accept4` reveals:

```
The accept4() system call is available starting with Linux 2.6.28.
Support in glibc is available starting with version 2.10.
```

So, your glibc must be `>=2.10` but your kernel is `<2.6.28`, thus the issue. `autoconf` detects that your glibc has `accept4()` so netdata builds without error, but your kernel does not have it, so it fails.

**Comment by ktsaou** [comment]
Did you solve it?
I am closing it, since I can't see a netdata issue here. If you believe there is something, please post.
